pkg/agent/controller/ipseccertificate/ipsec_certificate_controller.go

```diff
@@ -0,0 +1,403 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipseccertificate
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	certutil "k8s.io/client-go/util/cert"
+	csrutil "k8s.io/client-go/util/certificate/csr"
+	"k8s.io/client-go/util/keyutil"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	antreaapis "antrea.io/antrea/pkg/apis"
+	"antrea.io/antrea/pkg/ovs/ovsconfig"
+)
+
+const (
+	controllerName = "AntreaAgentIPsecCertificateController"
+	workerItemKey  = "key"
+	minRetryDelay  = 5 * time.Second
+	maxRetryDelay  = 300 * time.Second
+
+	// the mount path for CA certificate in antrea-ipsec container.
+	// StrongSwan will never reads CA certificates from folders other than `/etc/ipsec.d/cacerts`.
+	// Though StrongSwan will automatically load CA certificates from the folder, we set the ca_path in other_configs
+	// to the correct path for better consistency.
```

Comment:
During the tests i found that StrongSwan will read the cert and key from `ipsec.conf` and `ipsec.secrets` rendered by ovs. But it will not load the CA cert in `ipsec.conf`. so we need to load the CA certs into `/etc/ipsec.d/certs/` for `antrea-ipsec` container

Reply:
Maybe worth filing an issue to OVS for this mismatched behavior

Reply:
ok. Maybe this is also an issue for strongswan as the [doc](https://wiki.strongswan.org/projects/strongswan/wiki/CASection) shows that configure cacert path using absolute path is supported.

>  cacert = \<path\>
defines a path to the CA certificate either relative to [/etc/ipsec.d/cacerts] or as an absolute path.